gci/management/commands/publish_tasks.py

```diff
@@ -0,0 +1,96 @@
+import pprint
+
+from django.core.management.base import BaseCommand
+
+from gci.client import GCIAPIClient
+from gci.config import get_api_key
+from gci.config import load_yaml
+from gci.gitorg import get_issue
+from gci.issues import get_projects
+from gci.task import get_tasks
+
+
+class Command(BaseCommand):
+    help = 'Publish GCI tasks'
+
+    def add_arguments(self, parser):
+        parser.add_argument('filename')
+        parser.add_argument('--repo')
+        parser.add_argument('-a', '--ask', dest='ask',
+                            action='store_true', default=False)
+        parser.add_argument('-f', '--force', dest='force',
+                            action='store_true', default=False)
+
+    def handle(self, *args, **options):
+        filename = options.get('filename')
+        repo = options.get('repo')
+        ask = options.get('ask')
+        force = options.get('force')
+
+        issue_config = load_yaml(filename)
+
+        if repo:
+            issue_config = dict((key, data)
+                                for (key, data) in issue_config.items()
+                                if key == 'global' or repo in key)
+
+        projects = dict(get_projects(issue_config, yield_levels=False))
+
+        client = GCIAPIClient(get_api_key('GCI'))
+        tasks = get_tasks(private=True)
+
+        for task_id, task in tasks.items():
+            name = task['name']
+
+            if 'status' in task:
+                publish_flag = task['status']
+                if publish_flag == 2:
+                    # TODO: unpublishing tasks if project disabled/blocked
+                    continue
+                assert publish_flag == 1
+
+            if not task['mentors'] or not task['tags']:
+                continue
+
+            url = task['external_url']
+            if not url:
+                print(f'{task_id} {name} has no url')
+                continue
+            issue = get_issue(url)
+            if not issue:
+                print(f'{task_id} {name} {url} not recognised')
+                continue
+
+            project = projects.get(issue.repository.full_name)
+
+            if not project:
+                print(f'{task_id} {url} project not recognised')
+                continue
+
+            disabled = project.get('disabled')
+            if disabled:
+                print(f'{task_id} {url} project disabled')
+                continue
+
+            repo_block = project.get('blocked')
+            if repo_block and issue.number != repo_block:
+                print(f'{task_id} {url} project blocked by {repo_block}')
+                continue
+
+            task['status'] = 2
+
+            if ask:
+                pprint.pprint(task)
+                try:
+                    answer = input(f'Publish task {task_id} (y/n): ')
+                except BaseException:
+                    return
+
+                if answer != 'y':
+                    continue
```

Comment:
Maybe
```python
answer = None
while (answer != 'y' and answer != 'n'):
    answer = input(f'Publish task {task_id} (y/n): ')
if answer == 'n':
    continue
```

Reply:
I think I'll use coala_utils' ask functions, unless we can find a better (more djangoish) solution